### DISCUSSION TITLE
Question regarding overfitting a single batch and the weight-tying scheme

### DISCUSSION THREAD

**Discussion by pavanpreet-gandhi** [karpathy/build-nanogpt#61] [Q&A] [answered]
Question regarding overfitting a single batch [00:56:42](https://www.youtube.com/watch?v=l8pRSuU81PU&t=3402s).

When I overfit a single batch without the "weight tying" between the token embedding tensor and the language model head, I can achieve a really low loss (just like in the video). However, with the "weight tying", the loss seems to bottom out around 1 - why does this happen?

In other words, why does the weight-tying seem to limit the models ability to overfit a single batch?

**Comment by qaflan** [comment, marked as answer]
I believe weight sharing is a form of regularization and for that reason it reduces overfitting. Weight sharing reduces the number of effective learnable parameters (which reduces overfitting) and also imposes constraints on the network weights which is what regularization methods do to prevent overfitting.

**Reply by pavanpreet-gandhi** [reply]
That makes sense, thanks!
